src/Sleep.js

```diff
@@ -2,8 +2,60 @@ class Sleep {
   constructor(sleepData) {
     this.sleepData = sleepData;
   }
-}
 
+  __getUserDataById(userID) {
+    return this.sleepData.filter(dataItem => {
+      if (userID === dataItem.userID) {
+        return dataItem;
+      }
+    });
+  }
+
+  __getUserDataByDate(userID, date) {
+    return this.sleepData.find(dataItem => {
+      if (userID === dataItem.userID && date === dataItem.date) {
+        return dataItem;
+      }
+    });
+  }
+
+  getAvgDailySleepByUserId(userID) {
+    const sleepDataByUserId = this.__getUserDataById(userID);
```

Comment:
I see what you mentioned earlier about DRYing up code in certain areas where you re-use these helper functions. We can refactor together at a later point but I think these are a really great start!